pcsx2/CDVD/CdRom.cpp

```diff
@@ -848,22 +858,25 @@ void cdrWrite1(u8 rt)
 
 		case CdlSetmode:
 			CDVD_LOG("Setmode %x", cdr.Param[0]);
-
 			cdr.Mode = cdr.Param[0];
 			cdr.Ctrl |= 0x80;
 			cdr.Stat = NoIntr;
+			cdr.OldSpeed = cdr.Speed;
+			cdr.Speed = 1 + ((cdr.Mode >> 7) & 0x1);
 			if (cdr.Mode & MODE_CDDA)
 			{
 				StopCdda();
-				cdvd.Type = CDVD_TYPE_CDDA;
+				cdvd.Type = CDVD_TYPE_PSCDDA;
 			}
 
-			cdvd.Speed = 1 + ((cdr.Mode >> 7) & 0x1);
-			setPs1CDVDSpeed(cdvd.Speed);
+			if (cdr.Mode & MODE_SPEED && cdr.OldSpeed != cdr.Speed)
```

Comment:
This seems pointless. From a quick glance, `setPs1CDVDSpeed()` will have no side effects if the speed does not change. Also, this expression as is will break when switching from double back to single speed (very common, e.g. for CDDA).

From my hardware tests on a PS1 console, setting the mode starts changing the speed immediately, and it takes approximately 0.8 seconds to stabilize at the new speed from 1x->2x, or ~1 second from 2x->1x. Ideally, this should be taken into consideration when seeking/reading after a speed change.